src/components/table-hoc/TableWithPagination.tsx

```diff
@@ -49,14 +46,13 @@ export const tableWithPagination = (config: ITableWithPaginationConfig & Partial
 
             const startingIndex = paginationState.pageNb * perPageState.perPage;
             const endingIndex = startingIndex + perPageState.perPage;
-            const slice = config.sliceData || defaultSliceData;
 
             return {
                 pageNb: paginationState.pageNb,
                 perPage: perPageState.perPage,
                 totalEntries: length,
                 totalPages: Math.ceil(length / perPageState.perPage),
-                data: config.isServer ? ownProps.data : ownProps.data && slice(ownProps.data, startingIndex, endingIndex),
+                data: config.isServer ? ownProps.data : ownProps.data && sliceData(ownProps.data, startingIndex, endingIndex),

```

Comment:
Im not 100% sure but if we add a selector for a calcul with a ownProps. we can avoid a sliceData if the ownProps.data do not change.